Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIr.kt

```diff
@@ -229,72 +231,73 @@ enum class VisibilityModifier {
     PRIVATE, PROTECTED, INTERNAL, PUBLIC
 }
 
-class ConstructorParameterStub(val name: String, val type: StubType, val qualifier: Qualifier = Qualifier.NONE) {
-    sealed class Qualifier {
-        class VAL(val overrides: Boolean) : Qualifier()
-        class VAR(val overrides: Boolean) : Qualifier()
-        object NONE : Qualifier()
-    }
-}
-
 class GetConstructorParameter(
-        val constructorParameterStub: ConstructorParameterStub
+        val constructorParameterStub: FunctionParameterStub
 ) : ValueStub()
 
 class SuperClassInit(
         val type: StubType,
         val arguments: List<ValueStub> = listOf()
 )
 
+// TODO: Consider unifying these classes.
 sealed class ClassStub : StubContainer(), StubElementWithOrigin, AnnotationHolder {
 
     abstract val superClassInit: SuperClassInit?
     abstract val interfaces: List<StubType>
     abstract val childrenClasses: List<ClassStub>
     abstract val companion : Companion?
+    abstract val classifier: Classifier
 
     class Simple(
-            val classifier: Classifier,
+            override val classifier: Classifier,
             val modality: ClassStubModality,
-            val constructorParameters: List<ConstructorParameterStub> = emptyList(),
+            constructors: List<ConstructorStub> = emptyList(),
+            methods: List<FunctionStub> = emptyList(),
             override val superClassInit: SuperClassInit? = null,
             override val interfaces: List<StubType> = emptyList(),
             override val properties: List<PropertyStub> = emptyList(),
             override val origin: StubOrigin,
             override val annotations: List<AnnotationStub> = emptyList(),
             override val childrenClasses: List<ClassStub> = emptyList(),
             override val companion: Companion? = null,
-            override val functions: List<FunctionalStub> = emptyList(),
             override val simpleContainers: List<SimpleStubContainer> = emptyList()
-    ) : ClassStub()
+    ) : ClassStub() {
+        override val functions: List<FunctionalStub> = constructors + methods
+    }
 
     class Companion(
+            override val classifier: Classifier,
+            constructors: List<ConstructorStub> = emptyList(),
```

Comment:
Why not synthesize the constructor here? Isn't it the same for all `companion object`s?